harvester_e2e_tests/integration/test_vm_functions.py

```diff
@@ -101,14 +153,414 @@ def test_minimal_vm(api_client, image_id, unique_vm_name, wait_timeout):
+                         ids=['cpu', 'mem', 'disk', 'mem-and-cpu', 'mem-cpu-and-disk'])
+def test_create_vm_no_available_resources(request, resource, api_client, image_id,
+                                          wait_timeout, unique_vm_name):
+    """Creates a VM with outlandish resources for varying elements (purposefully negative test)
+
+    Prerequisite:
+        Setting opensuse-image-url set to a valid URL for
+        an opensuse image.
+
+    Manual Test Doc(s):
+        - N/A
+
+    Args:
+        request (FixtureRequest): https://docs.pytest.org/en/7.1.x/_modules/_pytest/fixtures.html#FixtureRequest # noqa
+        resource (dict): dict of name(s) & value that can be deconstructed
+        api_client (HarvesterAPI): HarvesterAPI client
+        image_id (str): corresponding image_id from fixture
+        wait_timeout (int): seconds for wait timeout from fixture
+        unique_vm_name (str): string of unique vm name
+
+    Raises:
+        AssertionError: when vm can not be created, all vms should be allowed to be created
+
+    Steps:
+    1. build vm object specs for outlandish resource(s) under test
+    2. request to build the vm, assert that succeeds
+    3. check for conditions of guest not running and vm being unschedulable
+    4. delete vm and volumes
+
+    Expected Result:
+    - building vm with outlandish resource requests to be successful
+    - asserting that the status condition of the vm that is built to not be running
+    - asserting that the status condition of the vm that is built to be unschedulable
+    - assert deleting vm and volumes to be successful
+    """
+    unique_name_for_vm = f"{''.join(resource.keys())}-{unique_vm_name}"
+    overall_vm_obj = dict(cpu=1, mem=2, disk=10)
+    overall_vm_obj.update(resource)
+
+    vm = api_client.vms.Spec(overall_vm_obj['cpu'], overall_vm_obj['mem'])
+    vm.add_image("disk-0", image_id, size=overall_vm_obj.get('disk'))
+    code, data = api_client.vms.create(unique_name_for_vm, vm)
+    assert 201 == code, (code, data)
+
+    endtime = datetime.now() + timedelta(seconds=wait_timeout)
+    while endtime > datetime.now():
+        code, data = api_client.vms.get_status(unique_name_for_vm)
+        if 200 == code and len(data.get('status', {}).get('conditions', [])) > 1:
+            check_dict = {
+                'not_running': None,
+                'unschedulable': None
+            }
+            for status_condition in data.get('status', {}).get('conditions', []):
+                if status_condition.get('reason') == 'GuestNotRunning':
+                    check_dict['not_running'] = True
+                if status_condition.get('reason') == 'Unschedulable':
+                    check_dict['unschedulable'] = True
+            assert check_dict["not_running"] is True, "the vm shouldn't be running"
+            assert check_dict["unschedulable"] is True, "the vm shouldn't be schedulable"
```

Comment:
We should check all conditions, so:
```python
assert all(check_dict.values()), (
    "the VM miss condition:\n"
    " and".join(k for k, v in check_dict.items() if not v)
)
```

then we can optimized to readability:
```python
checks = dict(GuestNotRunning=False, Unschedulable=False)
for condition in data['status']['conditions']:
    if condition.get('reason') in checks:
        checks[condition['reason']] = True

assert all(checks.values()), (
    "The VM miss condition:\n"
    " and ".join(k for k, v in check_dict.items() if not v)
)
```

Reply:
I like the simplicity with that, I've gone ahead and refactored that portion :smile: